plugins/rapture/nexus-coreui-plugin/src/test/java/org/sonatype/nexus/coreui/RepositoryXOTest.groovy

```diff
@@ -0,0 +1,103 @@
+package org.sonatype.nexus.coreui
+
+import javax.validation.Validation
+import javax.validation.Validator
+
+import org.sonatype.nexus.validation.group.Create
+
+import spock.lang.Shared
+import spock.lang.Specification
+import spock.lang.Unroll
+
+class RepositoryXOTest
+    extends Specification
+{
+
+  @Shared
+  Validator validator = Validation.buildDefaultValidatorFactory().validator
+
+  static List INVALID_CHARACTERS = ['_', '#', ',', 'A', '*', ' ', '#', '\'', '\\', '/', '?', '<', '>', '|', ' ', '\r',
+                                    '\n', '\t', ',', '.'].asImmutable()
```

Comment:
I'm leaning towards making it simple and strict, since these are used as identifiers.  Don't care about non-english use of repository names.  We'll add a description field at some point in the future that folks can put whatever they want in, in any language, but the name should be rigid.